protocol.go

```diff
@@ -147,6 +147,24 @@ func (p *Conn) RemoteAddr() net.Addr {
 	return p.header.SourceAddr
 }
 
+// TCPConn returns is the underlying connection as TCPConn
+func (p *Conn) TCPConn() (conn *net.TCPConn, ok bool) {
```

Comment:
Instead of exposing one helper per connection type, I'd prefer to expose an `Unwrap` function that returns the nested connection. This would work for `net.Conn` types that aren't from the stdlib, and wouldn't put more work on callers:

```go
conn, ok := conn.Unwrap().(*net.TCPConn)
```

(The name `Unwrap` comes from the similar mechanism used by the `errors` package.)

What do you think?

Reply:
Ah, I see this was already suggested earlier: https://github.com/pires/go-proxyproto/pull/51#pullrequestreview-519867029

Reply:
Thing here is we now have code like the following https://github.com/pires/go-proxyproto/blob/master/header.go#L96 which does not, in this case, return the nested source and destination `net.Addr` and puts the onus on the consumer to do the casting themselves as suggested above. Therefore, I am accepting this change as is.

Thank you for your feedback, @emersion.

Reply:
The addresses coming from the header are unrelated to the underlying connection. I could build a proxy from TCP to WebSockets which forwards the TCP addresses via the PROXY protocol, for instance.